File: app/src/main/java/dev/bltucker/nanodegreecapstone/storydetail/DetailStoryDiffCallback.java
package dev.bltucker.nanodegreecapstone.storydetail;

import android.support.v7.util.DiffUtil;

import dev.bltucker.nanodegreecapstone.models.Comment;

class DetailStoryDiffCallback extends DiffUtil.Callback{
    private final Comment[] oldComments;
    private final Comment[] updatedComments;

    DetailStoryDiffCallback(Comment[] oldComments, Comment[] updatedComments) {
        this.oldComments = oldComments;
        this.updatedComments = updatedComments;
    }

    @Override
    public int getOldListSize() {
        return oldComments.length;
    }

    @Override
    public int getNewListSize() {
        return updatedComments.length;
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        Comment oldComment = oldComments[oldItemPosition];
        Comment newComment = updatedComments[newItemPosition];

        return oldComment.getCommentId() == newComment.getCommentId();
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        Comment oldComment = oldComments[oldItemPosition];
        Comment newComment = updatedComments[newItemPosition];

        return oldComment.getCommentText().equals(newComment.getCommentText());
    }
}

File: app/src/main/java/dev/bltucker/nanodegreecapstone/storydetail/DetailStoryDiffCallback.kt
package dev.bltucker.nanodegreecapstone.storydetail

import android.support.v7.util.DiffUtil

import dev.bltucker.nanodegreecapstone.models.Comment

internal class DetailStoryDiffCallback(private val oldComments: Array<Comment>,
                                       private val updatedComments: Array<Comment>) : DiffUtil.Callback() {

    override fun getOldListSize(): Int {
        return oldComments.size
    }

    override fun getNewListSize(): Int {
        return updatedComments.size
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        val oldComment = oldComments[oldItemPosition]
        val newComment = updatedComments[newItemPosition]

        return oldComment.getCommentId() == newComment.getCommentId()
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        val oldComment = oldComments[oldItemPosition]
        val newComment = updatedComments[newItemPosition]

        return oldComment.getCommentText() == newComment.getCommentText()
    }
}
